<launch>
  <!-- Launch file for Livox AVIA LiDAR -->
  <arg name="rviz" default="true" />

  <rosparam command="load" file="$(find fast_lio_sam)/config/avia.yaml" />

  <!-- feature-based -->
  <param name="feature_extract_enable" type="bool" value="0"/>
  <param name="point_filter_num" type="int" value="3"/>                <!-- 零点滤波阈值，推荐2~5 -->
  <param name="lidar_type" type="int" value="8"/>   
  <param name="max_iteration" type="int" value="6" />
  <!-- 单帧内，畸变纠正后降采样的分辨率 -->
  <param name="filter_size_surf" type="double" value="0.2" /> 
  <param name="filter_size_corner" type="double" value="0.2" /> 
  <param name="filter_size_map" type="double" value="0.3" />
  <param name="cube_side_length" type="double" value="100" />
  <param name="runtime_pos_log_enable" type="bool" value="0" />
  <param name="loopClosureEnableFlag" type="bool"  value="true"/> 
  <param name="preprocess/blind"  type="double" value="0.01"/>
  <param name="preprocess/z_min"  type="double" value="-1.0" />
  <param name="preprocess/z_max"  type="double" value="100.0"/>
  <param name="preprocess/scan_line" type="double" value="16"/>
  <param name="feature_extract_enable" type="bool" value="false"/> 
  <param name="mapping/extrinsic_est_en" type="bool" value="true"/> 
  <param name="publish/path_en" type="bool" value="false"/> 
  <param name="publish/scan_bodyframe_pub_en" type="bool" value="false"/> 
  <param name="publish/scan_publish_en" type="bool" value="false"/>  <!--  是否发布点云-->
  <param name="publish/dense_publish_en" type="bool" value="false"/>  
  <param name="recontructKdTree" type="bool" value="false"/>
  <param name="visulize_IkdtreeMap" type="bool" value="false"/>
  <param name="yaw_angle" type="double" value="60.0" />
  
  <node pkg="fast_lio_sam" type="fastlio_sam_mapping" name="laserMapping" output="screen" /> 

  <arg name="map" default="" />
  
  <!-- loalization -->
  <node pkg="fast_lio_sam" type="global_localization.py" name="global_localization" output="screen" />

  <!-- transform fusion -->
  <node pkg="fast_lio_sam" type="transform_fusion.py" name="transform_fusion" output="screen" />

  <node pkg="pcl_ros" type="pcd_to_pointcloud" name="map_publisher" output="screen"
    args="/home/tuo/chatp4/chapt_ws/11.pcd 5 _frame_id:=map cloud_pcd:=/map" />
  

  <node pkg="fast_lio_sam" type="path_record.py" name="path_record" output="screen">
    <!-- 运行模式: click(点击模式)或record(记录模式) rosservice call /add_point "x: 0.0 y: 6.0"-->
    <param name="mode" value="click"/>
    <!-- 是否自动加载路径 -->
    <param name="auto_load" value="false" />
    <!-- 记录模式下的最小记录距离 (米) -->
    <param name="min_distance" value="0.2" />
    <!-- 路径平滑因子 -->
    <param name="smooth_factor" value="10.0" />
    <param name="goal_threshold" value="0.0"/>
    <!-- 里程计话题重映射 -->
    <remap from="/Odometry" to="/Odometry" />
  </node>
  
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_sam)/rviz_cfg/loam_livox.rviz" />
  </group>
</launch>